mico-core/src/main/java/io/github/ust/mico/core/broker/MicoServiceInterfaceBroker.java

```diff
@@ -1,4 +1,96 @@
 package io.github.ust.mico.core.broker;
 
+import io.fabric8.kubernetes.api.model.LoadBalancerIngress;
+import io.fabric8.kubernetes.api.model.LoadBalancerStatus;
+import io.github.ust.mico.core.exception.MicoServiceInterfaceAlreadyExistsException;
+import io.github.ust.mico.core.exception.MicoServiceInterfaceNotFoundException;
+import io.github.ust.mico.core.exception.MicoServiceNotFoundException;
+import io.github.ust.mico.core.model.MicoService;
+import io.github.ust.mico.core.model.MicoServiceInterface;
+import io.github.ust.mico.core.persistence.MicoServiceInterfaceRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Service
 public class MicoServiceInterfaceBroker {
+
+    @Autowired
+    private MicoServiceInterfaceRepository serviceInterfaceRepository;
+
+    @Autowired
+    private MicoServiceBroker micoServiceBroker;
+
+    public List<MicoServiceInterface> getInterfacesOfService(String shortName, String version) {
+        List<MicoServiceInterface> serviceInterfaces = serviceInterfaceRepository.findByService(shortName, version);
+
+        return serviceInterfaces;
+    }
+
+    public MicoServiceInterface getInterfaceOfServiceByName(String shortName, String version, String interfaceName) throws MicoServiceInterfaceNotFoundException {
+        Optional<MicoServiceInterface> micoServiceInterfaceOptional = serviceInterfaceRepository.findByServiceAndName(shortName, version, interfaceName);
+
+        if (!micoServiceInterfaceOptional.isPresent()) {
+            throw new MicoServiceInterfaceNotFoundException(shortName, version, interfaceName);
+        }
+
+        return micoServiceInterfaceOptional.get();
+    }
+
+    public List<String> getPublicIpsOfInterfaceByInterfaceName(String shortName, String version, String serviceInterfaceName, io.fabric8.kubernetes.api.model.Service kubernetesService) {
```

Comment:
This method is currently not used because you are using the implementation of `MicoStatusService`.
I recommend to open a new issue to refactor the `MicoStatusService` → `MicoStatusService` should not use the repositories `MicoServiceRepository`, `MicoServiceInterfaceRepository` and `MicoApplicationRepository` directly. Instead it should use the broker instances.
Proposal about the end result:
* `MicoServiceInterfaceBroker` includes all business logic regarding the interface resource (including the public IP stuff)
* `MicoStatusService` includes the logic for retrieving data from Prometheus and combining the data from various sources (`MicoServiceInterfaceBroker` and others) to provide a DTO for the API.

What do you think?
If you open such an issue, please add a comment to this method and reference the new issue.